torchao/prototype/moe_training/tensor.py

```diff
@@ -27,7 +27,7 @@
     torch.ops.aten.copy_.default,
     torch.ops.aten.view.default,
     torch.ops.aten.as_strided.default,
-    torch.ops.aten._to_copy.default,
+    torch.ops.aten._to_copy.default,  # for *.to(dtype)
```

Comment:
it seems a bit strange to me that `_to_copy` should **always** preserve the subclass. For example - if I do `aten._to_copy(mxfp8_tensor, dtype=torch.bfloat16)`, I would expect the output to **not** be a subclass, right?

Put another way - I'd imagine that you need to carefully implement `_to_copy` to only preserve the subclass if we are not doing a cross-dtype copy to a higher precision